bitbucket/resource_repository.go

```diff
@@ -154,6 +183,46 @@ func newRepositoryFromResource(d *schema.ResourceData) *bitbucket.Repository {
 	return repo
 }
 
+type forkWorkspace struct {
+	Slug string `json:"slug,omitempty"`
+}
+
+type forkedRepositoryBody struct {
+	Name string `json:"name,omitempty"`
+	Language string `json:"language,omitempty"`
+	IsPrivate bool `json:"is_private,omitempty"`
+	Description string `json:"description,omitempty"`
+	ForkPolicy string `json:"fork_policy,omitempty"`
+	HasWiki bool `json:"has_wiki,omitempty"`
+	HasIssues bool `json:"has_issues,omitempty"`
+	Links *bitbucket.RepositoryLinks `json:"links,omitempty"`
+	Project *bitbucket.Project `json:"project,omitempty"`
+	Workspace *forkWorkspace `json:"workspace,omitempty"`
+}
+
+func createForkedRepositoryFromRepository(repo *bitbucket.Repository, targetWorkspaceSlug string) *forkedRepositoryBody{
+	forkedRepo := &forkedRepositoryBody{
+		Name: repo.Name,
+		Language: repo.Language,
+		IsPrivate: repo.IsPrivate,
+		Description: repo.Description,
+		ForkPolicy: repo.ForkPolicy,
+		HasWiki: repo.HasWiki,
+		HasIssues: repo.HasIssues,
+	}
+	workspace := &forkWorkspace{
```

Comment:
why not reuse Repository struct? why is it needed to create another one

Reply:
The fork requires an additional field, workspace; unfortunately this doesn't seem to be documented in the OpenAPI spec (or rather, it is; "Extra properties of any type may be provided to this object.", to quote the API docs), in a way that the OpenAPI spec doesn't quite capture.